Mono repo take three

garrensmith opened the pull request:
This is my third and final push on getting this mono-repo. I've imported express-server in with full git history.

garrensmith commented:
Three things left to do:

* Tests must pass on Travis
* Update readme's
* Update package.json
* Create release scripts

nolanlawson commented:
We also need to put `pouchdb-server` package into `packages/node_modules`.

The basic idea is that the top-level `package.json` is a dummy package that never gets published. It just holds the dependencies so that Greenkeeper can update them. Then at publish time we scan for dependencies using `find-requires` and then inject those into the sub-package.jsons.

If I have time I'll work on this but it looks like a 1-2 full days effort at least (judging based on the time I put into the pouchdb monorepo).

garrensmith commented:
I made some nice progress on this and am almost done. Tests are all passing. I've updated readme's and package.json. Pouchdb-server was actually added to the file structure on my machine as a subpackage but for some reason it wasn't added to the repo. 

At this stage I think the only thing that needs to be done is add the release functionality. I'm hoping @nolanlawson can help me with that otherwise I will take a look next week. Once we have done that i would like to merge this and then move the issues and PR's from express.

Also cc @yaronyg and @gr2m to let them know about this since they have both expressed interest in this before.

gr2m commented on README.md:
```diff
@@ -1,195 +1,24 @@
-Please make your changes on a separate branch whose name reflects your changes, push them to your fork, and open a pull request!
-
-For commit message style guidelines, please refer to [PouchDB CONTRIBUTING.md](https://github.com/pouchdb/pouchdb/blob/master/CONTRIBUTING.md).
+See the CONTRIBUTING.md file for how to get involved
```
maybe link the CONTRIBUTING.md file? `[CONTRIBUTING.md](CONTRIBUTING.md)` should work I think

nolanlawson replied:
:+1:

nolanlawson commented:
OK, I have cleaned up the structure a little bit, implemented a release script, and tested it on Sinopia to make sure everything is working correctly. Release instructions are now in `CONTRIBUTING.md`.

Once this is green, I will merge and release 2.3.0 for both packages so we can get this in fast. (Note that we don't need to release 3.0.0 because there are no breaking changes here.)

garrensmith commented:
Awesome. Thanks for taking it over the finish line. 

All misspelling thanks to my iPhone.




On Sat, Feb 18, 2017 at 6:32 PM +0200, "Nolan Lawson" <notifications@github.com> wrote:












OK, I have cleaned up the structure a little bit, implemented a release script, and tested it on Sinopia to make sure everything is working correctly. Release instructions are now in CONTRIBUTING.md.


Once this is green, I will merge and release 2.3.0 for both packages so we can get this in fast. (Note that we don't need to release 3.0.0 because there are no breaking changes here.)



—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

nolanlawson commented:
No prob. Last bit is just getting the README cleaned up. I'll do that before publishing.

nolanlawson commented:
Done. 2.3.0 are published for both `express-pouchdb` and `pouchdb-server`. Thanks everyone!

gr2m commented:
great work @garrensmith @nolanlawson 🎉

nolanlawson commented:
Had a few publishing snafus, but 2.3.3 should be good now. Fourth time's the charm. 😅

garrensmith commented:
Woohoo. Thanks Nolan

All misspelling thanks to my iPhone.




On Sat, Feb 18, 2017 at 9:52 PM +0200, "Nolan Lawson" <notifications@github.com> wrote:












Had a few publishing snafus, but 2.3.3 should be good now. Fourth time's the charm. 😅



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.
